File: src/2025-03-04-12-20-21--deepseek-vr-0304/java/exercises/practice/transpose/src/main/java/Transpose.java
public class Transpose {
    public String transpose(String toTranspose) {
        if (toTranspose.isEmpty()) {
            return "";
        }
        
        String[] lines = toTranspose.split("\n");
        int maxLength = 0;
        for (String line : lines) {
            if (line.length() > maxLength) {
                maxLength = line.length();
            }
        }
        
        List<String> transposed = new ArrayList<>();
        for (int i = 0; i < maxLength; i++) {
            StringBuilder sb = new StringBuilder();
            for (String line : lines) {
                if (i < line.length()) {
                    sb.append(line.charAt(i));
                } else {
                    sb.append(' ');
                }
            }
            String row = sb.toString();
            // Trim trailing spaces
            int end = row.length();
            while (end > 0 && row.charAt(end - 1) == ' ') {
                end--;
            }
            row = row.substring(0, end);
            transposed.add(row);
        }
        return String.join("\n", transposed);
    }
}

File: src/python-2025-02-24-deepclaude-rust/python/exercises/practice/transpose/transpose.py
def transpose(text):
    lines = text.split('\n')
    max_len = max(len(line) for line in lines) if lines else 0
    transposed = []
    for i in range(max_len):
        row = []
        for line in lines:
            if i < len(line):
                row.append(line[i])
            else:
                row.append(' ')
        transposed_row = ''.join(row).rstrip()
        transposed.append(transposed_row)
    return '\n'.join(transposed)
